rocketpy/mathutils/function.py

```diff
@@ -19,12 +21,12 @@ class Function:
 
     def __init__(
         self,
-        source,
```

Comment:
Type hints are not supported yet (although I like your approach), so I'm afraid we need to remove the hints before merging.

Reply:
I didn't understand which errors you want to change, but feel free remove the ones that are not supported

Reply:
Type hinting (https://realpython.com/lessons/type-hinting/) is a Python feature that we didn't adopt in RocketPy yet. We had some discussions on top of #444 recently.

Example:
![image](https://github.com/RocketPy-Team/RocketPy/assets/63590233/f1ef11bb-6c25-4d26-9ee9-1f5b09e98cf6)

I'm removing the hints and annotations from this PR.  Again, I appreciate the intention, but we decided for not accepting hinting in this library yet, this will make maintainers life easier.